### PR TITLE
Fix index error

### PR DESCRIPTION
**Pull request by MaxNoe**
This is fixing #13  without changing default behaviour.

The methods now have a `fallback` argument which if True results in suitable values if the desired entry could not be found (`float('nan')` for floats, `None` for everything else).

### CODE REVIEW

**Comment by relleums** [comment]
Have you added the default and fallback feature only for the disk spaces?
What was the initial motivation behind the changes?

**Comment by MaxNoe** [comment]
No, fallback is used everywhere if `fallback=True`, this is achieved by using `partial(get_entry, fallback=fallback`).

The default value for the fallback is `None`, only for the disk spaces I need a different fallback, because we expect a tuple of (value, unit).

**Comment by MaxNoe** [comment]
The motivation is discussed in the linked issue #13

File: smart_fact_crawler/__init__.py [hunk 1]
```diff
@@ -39,252 +43,306 @@ def smartfact(timeout=None):
...
         new_date = now.replace(hour=hour, minute=minute, second=0)
         if not new_date > now:
             new_date += timedelta(days=1)
+
         return new_date
 
     conv = next_datetime_from_hhmm_string
 
     table = smartfact2table(url, timeout=timeout)
+    get = partial(get_entry, fallback=fallback)
```
Comment (dneise) [inline smart_fact_crawler/__init__.py]: Its a pity there is so much code repetition here now ... this:
```
   get = partial ..
```
... is everywhere.

File: smart_fact_crawler/__init__.py [hunk 2]
```diff
@@ -39,252 +43,306 @@ def smartfact(timeout=None):
...
-        'timestamp': sft2dt(table[0][0]),
-        'end_of_dark_time': conv(table[1][1]),
-        'end_of_astronomical_twilight': conv(table[2][1]),
-        'end_of_nautical_twilight': conv(table[3][1]),
-        'start_of_day_time': conv(table[4][1]),
-        'end_of_day_time': conv(table[5][1]),
-        'start_of_nautical_twilight': conv(table[6][1]),
-        'start_of_astronomical_twilight': conv(table[7][1]),
-        'start_of_dark_time': conv(table[8][1]),
+        'timestamp': sft2dt(get(table, 0, 0)),
```
Comment (dneise) [inline smart_fact_crawler/__init__.py]: You said somewhere the `get(table, row, col)` is meant to work just like `dict.get(key)`.

I was wondering, why you didn't make `get` a method of a `SmartfactTable` class, which might be returned by `smartfact2table()`. In that case one could let the `fallback` parameter be a parameter for the ctor of `SmartfactTable`

I think this does not change anything here with respect to lines of code or readability.... I was just curious.

Comment (MaxNoe) [inline smart_fact_crawler/__init__.py]: That would adding a new class, which I didn't think about. One could do that, but I'm also fine with a function that acts on built in lists

File: smart_fact_crawler/__init__.py (hunk 1 above)
Comment (MaxNoe) [inline smart_fact_crawler/__init__.py]: Yeah, true. But kind of unavoidable I think, as I do not want to set fallback globally

**Comment by MaxNoe** [comment]
We could think about setting `fallback=True` as default, I was a bit reluctant to do that. But as we are basically the only users and the shifthelper runs on a fixed version number there is no big harm to be done I guess.

**Comment by dneise** [comment]
I cannot form a well based opinion on this matter .. do it the way you like it best.

**Comment by MaxNoe** [comment]
Awesome

**Pull request merged**